### ISSUE TITLE
icefall freezes at the end of epoch zero

### ISSUE THREAD

**Issue by fra1993** [k2-fsa/k2#847]
Hi, I am running k2 on a HPC facility. I have installed k2, lhotse and icefall inside a singularity image with conda and following example for tdnn the training starts (in interactive shell I can see the logs). However, I think at the end of the first epoch (batch 150000), the training freezes. Any Ideas about this behavior? Thank you in advance.

**Comment by csukuangfj** [comment]
So you are training with a single GPU since you don't specify "--world-size". What do you mean by "freeze"? Does it get stuck during saving checkpoints?

**Comment by csukuangfj** [comment]
Can you check that the process with rank 0 can save checkpoints to disk without problems? 

You can do an early return in the training loop to see whether it can save checkpoints. 

E.g., in the training loop, do

"""
if batch_idx > 50:
  break
"""

**Comment by fra1993** [comment]
Ok this worked and I was able to get the checkpoints. I think the problem I had was that on the node I was using I requested only one cpu. With 8 cpus the training works perfectly and now I am trying the whole training (without stopping at batch 50). I am however getting a warning: `WARNING:root:To perform mix, energy must be non-zero and non-negative (got 0.0).  MonoCut with id "a417a556-e4d6-439a-8e66-48e34981edab" will not be mixed in.`
